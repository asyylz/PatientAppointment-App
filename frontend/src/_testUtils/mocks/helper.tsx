import { within } from '@testing-library/react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { Store } from '@reduxjs/toolkit';
import React from 'react';

export const toContainRole = async (
  container: HTMLElement,
  role: string,
  quantity = 1
): Promise<{ pass: boolean; message: () => string }> => {
  const elements = await within(container).findAllByRole(role);


  if (elements.length === quantity) {
    return {
      pass: true,
      message: () => '', // return an empty string when the test passes
    };
  }

  return {
    pass: false,
    message: () =>
      `Expected to find ${quantity} ${role} elements. Found ${elements.length} instead.`,
  };
};

export const renderComponent = (
  component: React.ReactElement,
  store: Store,
  options?: RenderOptions
) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>{component}</MemoryRouter>
    </Provider>,
    options
  );
};

// export const toContainRole = (
//   container: HTMLElement,
//   role: string,
//   quantity = 1
// ): { pass: boolean; message: () => string } => {
//   const elements = within(container).getAllByRole(role);
// console.log(elements.length)
//   if (elements.length === quantity) {
//     return {
//       pass: true,
//       message: () => '', // return an empty string when the test passes
//     };
//   }

//   return {
//     pass: false,
//     message: () =>
//       `Expected to find ${quantity} ${role} elements. Found ${elements.length} instead.`,
//   };
// };
