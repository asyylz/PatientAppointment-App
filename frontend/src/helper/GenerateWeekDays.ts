// Function to get day name from index (0 to 6)
const getDayName = (index: number) => {
  const days = [
    'Sunday',
    'Monday',
    'Tuesday',
    'Wednesday',
    'Thursday',
    'Friday',
    'Saturday',
  ];
  return days[index];
};

// Function to generate dates dynamically
export const generateDates = (numberOfDays: number) => {
  const dates = [];
  const today = new Date(); // Get today's date

  for (let i = 0; i < numberOfDays; i++) {
    const date = new Date(today.getTime() + i * 24 * 60 * 60 * 1000); // Add i days to today
    const dayName = getDayName(date.getDay()); // Get day name (e.g., "Monday")

    // Format date as "dd/mm/yyyy"
    const formattedDate = `${date.getDate()}/${
      date.getMonth() + 1
    }/${date.getFullYear()}`;

    // Push formatted date and day name to array
    dates.push({
      date: formattedDate,
      day: dayName,
    });
  }

  return dates;
};

// Example usage: Generate dates for the next 7 days
// const numberOfDays = 7;
// const dates = generateDates(numberOfDays);

// console.log(dates);
