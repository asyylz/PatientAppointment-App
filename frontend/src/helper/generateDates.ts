const days = [
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday',
  'Sunday',
];

export const getWeekDatesFromToday = () => {
  const today = new Date();
  const currentDayIndex = today.getDay(); // 0 is Sunday, 1 is Monday, etc.
  const startOfWeek = new Date(today);

  // Adjust to start the week from the current day
  startOfWeek.setDate(
    today.getDate() - (currentDayIndex === 0 ? 6 : currentDayIndex - 1)
  );

  // Function to add days to a date
  const addDays = (date: Date, days: number) => {
    const newDate = new Date(date);
    newDate.setDate(date.getDate() + days);
    return newDate;
  };

  const weekDates = days.map((day, index) => {
    const date = addDays(startOfWeek, index);
    const formattedDate = `${date.getDate().toString().padStart(2, '0')}/${(
      date.getMonth() + 1
    )
      .toString()
      .padStart(2, '0')}/${date.getFullYear()}`;
    return { date: formattedDate, day };
  });

  return weekDates;
};

export const convertDateAndTimeStringToDate = (
  dateString: string,
  timeString: string
): Date => {
  // Split time string into hours and minutes
  const [hours, minutes] = timeString.split(':').map(Number);

  // Define a variable to hold the new Date
  let newDate: Date;

  // Check the format of the date string and create a new Date object accordingly
  if (dateString.includes('/')) {
    const [day, month, year] = dateString.split('/').map(Number);
    newDate = new Date(year, month - 1, day, hours + 1, minutes, 0);
  } else {
    const [year, month, day] = dateString.split('-').map(Number);
    newDate = new Date(year, month - 1, day, hours + 1, minutes, 0);
  }

  return newDate;
};

export const formatDateForInput = (dateString: string): string => {
  //console.log(dateString);
  const parts = dateString.split('/');
  return `${parts[2]}-${parts[1]}-${parts[0]}`;
};

export const getDateFromDateString = (dateString: string): string => {
  console.log(dateString);
  const part = dateString.split('T');
  const parts = part[0].split('-');
  return `${parts[2]}/${parts[1]}/${parts[0]}`;
};

export const convertToDateandDateString = (
  availabilityDay: string,
  availabilityTime: string
) => {
  // Map days of the week to their index
  const daysOfWeek = {
    Sunday: 0,
    Monday: 1,
    Tuesday: 2,
    Wednesday: 3,
    Thursday: 4,
    Friday: 5,
    Saturday: 6,
  };

  // Get the current date
  const currentDate = new Date();
  // Get the current week's Monday
  const currentMonday = new Date(
    currentDate.setDate(currentDate.getDate() - currentDate.getDay() + 1)
  );

  // Adjust the day of the week to get the specific day's date
  const availabilityDate = new Date(currentMonday);
  availabilityDate.setDate(
    currentMonday.getDate() +
      (Object.keys(daysOfWeek).indexOf(availabilityDay) - 1 || 0)
  );

  // Combine the availability date and time into a single string
  const availabilityDateTimeString = `${
    availabilityDate.toISOString().split('T')[0]
  }T${availabilityTime}:00.000Z`;

  // Create a Date object for the availability
  const availabilityDateTime = new Date(availabilityDateTimeString);

  //console.log(availabilityDateTimeString);
  // Create a Date object for the current date and time
  const currentDateTime = new Date();

  // Compare the availability date and time with the current date and time
  return { availabilityDateTime, availabilityDateTimeString };
};

/* ---------------------- separate ---------------------- */

export const formatDateForUI = (isoString: string) => {
  console.log(isoString);
  // Extract the date part from the ISO string
  const datePart = isoString.split('T')[0];
  const [year, month, day] = datePart.split('-').map(Number);

  // Create a date object using the extracted date components
  const date = new Date(Date.UTC(year, month - 1, day)); // Date.UTC to avoid time zone conversion

  const options = { day: '2-digit', month: '2-digit', year: 'numeric' };
  const formattedDate = date.toLocaleDateString('en-GB', options); // Format as DD/MM/YYYY

  const daysOfWeek = [
    'Sunday',
    'Monday',
    'Tuesday',
    'Wednesday',
    'Thursday',
    'Friday',
    'Saturday',
  ];
  const dayOfWeek = daysOfWeek[date.getUTCDay()]; // Get the day of the week

  return `${formattedDate} ${dayOfWeek}`;
};
