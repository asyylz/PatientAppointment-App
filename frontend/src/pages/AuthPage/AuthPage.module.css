@import url('./../../App.css');

.container {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    align-items: center;
    justify-items: center;
    color: #707070;
    min-height: 100vh;
    background: #F2F2F2;
    -webkit-animation: moveFromLeftFade 1s ease both;
    animation: moveFromLeftFade 1s ease both;

}

.container::before {
    content: '';
    position: absolute;
    left: 50vw;
    top: 0;
    height: 100vh;
    width: 1px;
    border-radius: 12px;
    background: #C7BEBE;
}


.wrapper {
    position: relative;
    max-width: 450px;
    width: 40vw;
    background: #fff;
    padding: 34px;
    border-radius: 6px;
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
}


.wrapper h2 {
    position: relative;
    font-size: clamp(18px, 1.7vw, 2.5vw);
    font-weight: 600;
    color: #707070;
}

.wrapper h2::before {
    content: '';
    position: absolute;
    left: 0;
    bottom: -10px;
    height: 1px;
    width: 100%;
    border-radius: 12px;
    background: #C7BEBE;
}

.wrapper form {
    margin-top: 30px;
}


form .policy__wrapper {
    display: flex;
    align-items: center;
    margin-top: 1.5rem;
    justify-content: start;
}

.policy__wrapper h3 {
    font-size: clamp(10px, 1.7vw, 14px);
    margin: auto 0;
    margin-left: 6px;
    font-weight: bold;
    color: #707070;

}



.policy__wrapper input {
    height: 15px;
    width: 15px;
    margin: auto 0;
    min-width: 30px;
}



.wrapper button,
.imagePickerWrapper {
    margin-top: 1rem;
}

.wrapper button {
    width: 50%;
}

.forgot-password__wrapper {
    margin: 2rem 0;
    display: flex;
    gap: 1rem;
}

.forgot-password__wrapper a {
    color: #444;
    text-decoration: underline;
    cursor: pointer;
}

.forgot-password__wrapper a:hover {
    scale: 1.2;
}

.forgot-password__wrapper button {
    color: #444;
    text-decoration: underline;
    cursor: pointer;
    width: 50%;
}

.forgot-password__wrapper p,
.forgot-password__wrapper a {
    font-size: clamp(12px, 1.7vw, 14px);
}

@media (min-width:320px) and (max-width:767px) {
    .container {
        grid-template-columns: 1fr;
    }

    .wrapper {
        width: 80vw;
        margin: 2rem auto;
    }

    form button {
        font-size: 10px;
        width: 120px;
        height: 30px;
    }
}


@media (max-width: 767px) {
    .container::before {
        display: none;
    }
}