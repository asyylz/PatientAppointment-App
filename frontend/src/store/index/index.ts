import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import departmentsReducer from '../departments-slice/departments-slice';
import doctorsReducer from '../doctors-slice/doctors-slice';
import reviewsReducer from './../reviews-slice/reviews-slice';
import currentUserReducer from '../currentUser-slice/currentUser-slice';
import searchReducer from '../search-slice/search-slice';
import appointmentsForDoctorReducer from '../appointmentsForDoctor-slice/appointmentsForDoctor-slice';
import appointmentsForPatientReducer from '../appointmentsForPatient-slice/appointmentsForPatient-slice';
import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web

const authPersistConfig = {
  key: 'currentUser',
  storage: storage,
  // blacklist: ['status', 'error'],
};

const doctorPersistConfig = {
  key: 'doctors',
  storage: storage,
  // blacklist: ['status','error']
};

export const allReducers = combineReducers({
  currentUser: persistReducer(authPersistConfig, currentUserReducer),
  departments: departmentsReducer,
  doctors: persistReducer(doctorPersistConfig, doctorsReducer),
  reviews: reviewsReducer,
  search: searchReducer,
  appointmentsForDoctor: appointmentsForDoctorReducer,
  appointmentsForPatient: appointmentsForPatientReducer,
});

const store = configureStore({
  reducer: allReducers,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [
          FLUSH,
          REHYDRATE,
          PAUSE,
          PERSIST,
          PURGE,
          REGISTER,
          'appointmentsForPatient/fetchWithId/rejected',
          'appointmentsForDoctor/fetchWithId/rejected',
          'departments/fetch/rejected',
          'doctors/fetch/rejected',
        ],
      },
    }),
  devTools: process.env.NODE_ENV !== 'production',
});

export function setupStore(reducer?: any, preloadedState?: Partial<RootState>) {
  //console.log('from index store', preloadedState); // i can see my preloaded state here  with mock data
  const store = configureStore({
    reducer: reducer,
    preloadedState,
  });

  //console.log(store.getState()); // but still cant see current user but can see the rest of the states
  return store;
}

export type AppDispatch = typeof store.dispatch;
export type AppStore = typeof store;
export type RootState = ReturnType<AppStore['getState']>;

export const persistor = persistStore(store);

export default store;
