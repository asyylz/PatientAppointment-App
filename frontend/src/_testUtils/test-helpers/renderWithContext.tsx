import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setupStore } from '../../store/index';
import type { AppStore, RootState } from '../../store/index';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  reducer?: any;
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  extendedRenderOptions: ExtendedRenderOptions = {}
) {
  const { preloadedState, reducer } = extendedRenderOptions;
  // Ensure that the store is being correctly initialized with preloadedState
  const store = setupStore(reducer, preloadedState);
  //console.log(store.getState()); // Verify the state of the store

  const Wrapper = ({ children }: PropsWithChildren) => (
    <Provider store={store}>{children}</Provider>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...extendedRenderOptions }),
  };
}
