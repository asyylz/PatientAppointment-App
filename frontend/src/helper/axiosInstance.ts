import axios, { AxiosInstance } from 'axios';
import store from './../store/index'; // Adjust the path to your store file

export const createAxiosInstance = (): AxiosInstance => {
  const { token } = store.getState().currentUser;
  console.log(token);
  console.log('asiye')

  return axios.create({
    baseURL: import.meta.env.BASE_URL,
    headers: { Authorization: `Bearer ${token}` },
  });
};
