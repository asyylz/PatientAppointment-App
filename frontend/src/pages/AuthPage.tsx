import { useSelector } from 'react-redux';
import classes from './AuthPage.module.css';
import { useDispatch } from 'react-redux';
import { AppDispatch } from './../store/index';
import { loginCurrentUser } from '../store/currentUser-slice';
import { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';

const AuthPage = () => {
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');

  const dispatch: AppDispatch = useDispatch();

  const navigate = useNavigate();
  const {
    entities: { token, data },
    status,
    error,
  } = useSelector((state: RootState) => state.currentUser);

  const handleLogin = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    dispatch(loginCurrentUser({ email, password }));
  };

  useEffect(() => {
    if (status === 'succeeded') {
      console.log(status);
      navigate('/');
    }
  }, [status, navigate]);

  return (
    <div className={classes.container}>
      {/*------------------------ Login ----------------------- */}
      <div className={classes.wrapper}>
        <h2>Login</h2>
        <form action="" onSubmit={handleLogin}>
          <div className={classes.inputBox}>
            <input
              type="text"
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              required
            />
          </div>
          <div className={classes.inputBox}>
            <input
              type="password"
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              required
            />
          </div>
          <div className={`${classes.inputBox} ${classes.button}`}>
            <input type="submit" value="Login" />
          </div>
        </form>
      </div>
      {/*------------------------ Register ----------------------- */}
      <div className={classes.wrapper}>
        <h2>Registration</h2>
        <form action="">
          <div className={classes.inputBox}>
            <input type="text" placeholder="Enter your name" required />
          </div>
          <div className={classes.inputBox}>
            <input type="text" placeholder="Enter your email" required />
          </div>
          <div className={classes.inputBox}>
            <input type="password" placeholder="Create password" required />
          </div>
          <div className={classes.inputBox}>
            <input type="password" placeholder="Confirm password" required />
          </div>
          <div className={classes.policy}>
            <input type="checkbox" />
            <h3>I accept all terms & condition</h3>
          </div>
          <div className={`${classes.inputBox} ${classes.button}`}>
            <input type="submit" value="Register Now" />
          </div>
        </form>
      </div>
    </div>
  );
};

export default AuthPage;
