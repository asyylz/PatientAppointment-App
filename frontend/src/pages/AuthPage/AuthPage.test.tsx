import '@testing-library/jest-dom';
import { fireEvent, screen, waitFor } from '@testing-library/react';
import AuthPage from './AuthPage';
import { toastSuccessNotify } from '../../helper/ToastNotify';
import { act } from 'react';
import { combineReducers } from 'redux';
import currentUserReducer from '../../store/currentUser-slice/currentUser-slice';
import * as router from 'react-router-dom';
import { renderWithProviders } from '../../_testUtils/test-helpers/renderWithContext';
import { AppStore } from '../../store/index/index';
//import * as currentUserSlice from '../store/currentUser-slice';

jest.mock('../../helper/ToastNotify', () => ({
  toastSuccessNotify: jest.fn(),
}));

// Mock useNavigate
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

jest.mock('redux-persist', () => {
  const actual = jest.requireActual('redux-persist');
  return {
    ...actual,
    persistReducer: jest.fn().mockImplementation((_config, reducer) => reducer),
  };
});

const reducers = combineReducers({
  currentUser: currentUserReducer,
});

describe('AuthPage', () => {
  let store: AppStore;
  let asFragment: () => DocumentFragment;
  let navigate: any;

  beforeEach(() => {
    navigate = jest.fn();
    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
    const result = renderWithProviders(<AuthPage />, {
      reducer: reducers,
    });
    store = result.store;
    asFragment = result.asFragment; // asFragment() is used to create snapshots, which is the React Testing Library equivalent of toJSON().
    jest.spyOn(store, 'dispatch');
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  /* -------------------------- - ------------------------- */
  it('1--Snapshot before and after policy checkbox interaction', async () => {
    // Initial snapshot
    expect(asFragment()).toMatchSnapshot();

    // Find and interact with the login button
    const checkBox = screen.getByRole('checkbox');
    expect(checkBox).toBeInTheDocument();

    // Simulate checkbox click
    fireEvent.click(checkBox);

    // Re-render and create a new snapshot
    expect(asFragment()).toMatchSnapshot();
  });

  /* -------------------------- - ------------------------- */
  it('1--Should render login form', () => {
    expect(screen.getAllByText('Login')[0]).toBeInTheDocument();
    expect(
      screen.getAllByPlaceholderText('Enter your email')[0]
    ).toBeInTheDocument();
    expect(
      screen.getAllByPlaceholderText('Enter your password')[0]
    ).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Login' })).toBeInTheDocument();
    expect(screen.getByText('Forgot your password?')).toBeInTheDocument();
    expect(screen.getByText('Click here')).toBeInTheDocument();
  });

  /* -------------------------- - ------------------------- */
  it('2--Should render registration form', () => {
    expect(screen.getByText('Registration')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Enter your name')).toBeInTheDocument();
    expect(
      screen.getAllByPlaceholderText('Enter your email')[1]
    ).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Enter your DOB')).toBeInTheDocument();
    expect(
      screen.getAllByPlaceholderText('Enter your password')[1]
    ).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText('Confirm your password')
    ).toBeInTheDocument();
    expect(screen.getByRole('checkbox')).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: 'Register' })
    ).toBeInTheDocument();
  });

  /* -------------------------- - ------------------------- */
  it('3--Should dispatch login action on form submission', async () => {
    expect(store.getState().currentUser.status).toEqual('idle');
    const emailInput = screen.getAllByPlaceholderText('Enter your email')[0];
    const passwordInput = screen.getAllByPlaceholderText(
      'Enter your password'
    )[0];

    const loginButton = screen.getByRole('button', { name: 'Login' });

    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    fireEvent.change(passwordInput, { target: { value: 'password123' } });
    fireEvent.click(loginButton);

    expect(store.getState().currentUser.status).toEqual('loading');
    await act(async () => {
      store.dispatch({
        type: 'currentUser/login/fulfilled',
        payload: {
          token: 'mock-token',
          data: {
            user: {
              _id: '6673662fbd42a966b75dec92',
              name: 'Asiye',
              email: 'alice@test.com',
              role: 'patient',
              __v: 0,
              DOB: '1986-01-22T00:00:00.000Z',
              image:
                'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1722605356602-asiye%20serife.jpeg',
            },
          },
        },
      });
    });
    expect(store.getState().currentUser.status).toEqual('login success');
    //console.log(store.getState());

    // await waitFor(async () => {
    //   expect(screen.getAllByPlaceholderText('Enter your email')[0]).toHaveValue(
    //     ''
    //   );
    //   expect(
    //     screen.getAllByPlaceholderText('Enter your password')[0]
    //   ).toHaveValue('');
    // });

    expect(navigate).toHaveBeenCalledWith('/user/dashboard');
  });

  /* -------------------------- - ------------------------- */
  it('4--Should dispatch register action on form submission', async () => {
    const nameInput = screen.getByPlaceholderText('Enter your name');
    const emailInput = screen.getAllByPlaceholderText('Enter your email')[1];
    const dobInput = screen.getByPlaceholderText('Enter your DOB');
    const passwordInput = screen.getAllByPlaceholderText(
      'Enter your password'
    )[1];
    const confirmPasswordInput = screen.getByPlaceholderText(
      'Confirm your password'
    );
    const acceptTermsCheckbox = screen.getByRole('checkbox');
    const registerButton = screen.getByRole('button', { name: 'Register' });

    fireEvent.change(nameInput, { target: { value: 'John Doe' } });
    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    fireEvent.change(dobInput, { target: { value: '1990-01-01' } });
    fireEvent.change(passwordInput, { target: { value: 'password123' } });
    fireEvent.change(confirmPasswordInput, {
      target: { value: 'password123' },
    });
    fireEvent.click(acceptTermsCheckbox);
    fireEvent.click(registerButton);

    expect(store.getState().currentUser.status).toEqual('loading');

    await act(async () => {
      store.dispatch({
        type: 'currentUser/register/fulfilled',
        payload: {
          token: 'mock-token-register',
          data: {
            user: {
              _id: '6673662fbd42a966b75dec92',
              name: 'Asiye',
              email: 'alice@test.com',
              role: 'patient',
              __v: 0,
              DOB: '1986-01-22T00:00:00.000Z',
              image:
                'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1722605356602-asiye%20serife.jpeg',
            },
          },
        },
      });
    });
    console.log(store.getState().currentUser.status);
    expect(store.getState().currentUser.status).toEqual('register success');

    // await waitFor(() => {
    //   expect(screen.getByPlaceholderText('Enter your name')).toHaveValue('');
    //   expect(screen.getAllByPlaceholderText('Enter your email')[1]).toHaveValue(
    //     ''
    //   );

    //   expect(
    //     screen.getAllByPlaceholderText('Enter your password')[1]
    //   ).toHaveValue('');
    //   expect(screen.getByPlaceholderText('Enter your DOB')).toHaveValue(
    //     '1990-01-01'
    //   );
    //   expect(screen.getByPlaceholderText('Confirm your password')).toHaveValue(
    //     ''
    //   );
    //   expect(screen.getByRole('checkbox')).not.toBeChecked();
    // });

    const message = 'Successfully registered!';
    toastSuccessNotify(message);
    expect(toastSuccessNotify).toHaveBeenCalledWith(message);
  });

  /* -------------------------- - ------------------------- */
  it('5--Should forgotPassword action send email', async () => {
    const emailInput = screen.getAllByPlaceholderText('Enter your email')[0];
    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    const forgotPasswordButton = screen.getByText('Click here');
    fireEvent.click(forgotPasswordButton);
    await waitFor(() => {
      expect(
        store.dispatch({
          type: 'currentUser/forgotPassword/fulfilled',
          payload: { email: 'test@example.com' },
        })
      );
    });

    const message = 'Email sent to test@example.com successfully!';
    toastSuccessNotify(message);
    expect(toastSuccessNotify).toHaveBeenCalledWith(message);
  });
});
