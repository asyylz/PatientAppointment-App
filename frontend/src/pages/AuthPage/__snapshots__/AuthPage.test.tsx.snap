// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage 1--Snapshot before and after policy checkbox interaction 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="wrapper"
    >
      <h2>
        Login
      </h2>
      <form
        aria-label="login-form"
      >
        <div
          class="wrapper input"
        >
          <input
            name="email"
            placeholder="Enter your email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="password"
            placeholder="Enter your password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="forgot-password__wrapper"
        >
          <p>
            Forgot your password?
          </p>
          <a>
            Click here
          </a>
        </div>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
    <div
      class="wrapper"
    >
      <h2>
        Registration
      </h2>
      <form>
        <div
          class="wrapper input"
        >
          <input
            name="name"
            placeholder="Enter your name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="email"
            placeholder="Enter your email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="DOB"
            placeholder="Enter your DOB"
            required=""
            type="date"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="password"
            placeholder="Enter your password"
            type="password"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="passwordConfirm"
            placeholder="Confirm your password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="policy__wrapper"
        >
          <input
            name="policy"
            required=""
            type="checkbox"
            value="false"
          />
          <h3>
            I accept all terms & condition
          </h3>
        </div>
        <button
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AuthPage 1--Snapshot before and after policy checkbox interaction 2`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="wrapper"
    >
      <h2>
        Login
      </h2>
      <form
        aria-label="login-form"
      >
        <div
          class="wrapper input"
        >
          <input
            name="email"
            placeholder="Enter your email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="password"
            placeholder="Enter your password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="forgot-password__wrapper"
        >
          <p>
            Forgot your password?
          </p>
          <a>
            Click here
          </a>
        </div>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
    <div
      class="wrapper"
    >
      <h2>
        Registration
      </h2>
      <form>
        <div
          class="wrapper input"
        >
          <input
            name="name"
            placeholder="Enter your name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="email"
            placeholder="Enter your email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="DOB"
            placeholder="Enter your DOB"
            required=""
            type="date"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="password"
            placeholder="Enter your password"
            type="password"
            value=""
          />
        </div>
        <div
          class="wrapper input"
        >
          <input
            name="passwordConfirm"
            placeholder="Confirm your password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="policy__wrapper"
        >
          <input
            name="policy"
            required=""
            type="checkbox"
            value="true"
          />
          <h3>
            I accept all terms & condition
          </h3>
        </div>
        <button
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
